sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosEndToEndOperationLatencyPolicyConfig.java

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.CosmosEndToEndOperationLatencyPolicyConfigBuilder;
+
+import java.time.Duration;
+
+/**
+ * Represents End to end operation latency policy config
+ * This enables requests to get cancelled by the client once the specified timeout is reached
+ */
+public class CosmosEndToEndOperationLatencyPolicyConfig {
+    /**
+     * The default end to end operation latency policy with timeout of 2 seconds
+     */
+    public static final CosmosEndToEndOperationLatencyPolicyConfig DEFAULT =
+        new CosmosEndToEndOperationLatencyPolicyConfigBuilder().build();
+
+    /**
+     * The disabled end to end operation latency policy
+     */
+    public static final CosmosEndToEndOperationLatencyPolicyConfig DISABLED =
+        new CosmosEndToEndOperationLatencyPolicyConfigBuilder(false).build();
+
+    private final boolean isEnabled;
+    private final Duration endToEndOperationTimeout;
+
+    /**
+     * Constructor
+     * @param isEnabled toggle if the policy should be enabled or disabled
+     * @param endToEndOperationTimeout the timeout for request cancellation in {@link Duration}. Setting very low timeouts
+     *                                 can cause the request to never succeed.
+     */
+    public CosmosEndToEndOperationLatencyPolicyConfig(
+        boolean isEnabled,
+        Duration endToEndOperationTimeout) {
+
+        this.isEnabled = isEnabled;
+        this.endToEndOperationTimeout = endToEndOperationTimeout;
+    }
+
+    /**
+     * Returns if the policy is enabled or not
+     *
+     * @return if the policy is enabled or not
+     */
+    public boolean isEnabled() {
```

Comment:
I wonder should we allow customer to disable/enable it at any time.

one scenario I am thinking is:
customer can configure the timeout policy, in case some issue happens and they want to disable the policy, one way is to code change and remove the timeout policy reference, the other pretty common way is they kind have some BRS to turn off it directly(this will avoid a code deployment)

so in short words, my questions is can this config be dynamically configured

Reply:
No - overly complex and I think it is ok to request process restart in that case

Reply:
Customers will want to change that on the fly - no new code deployments. If it is not possible/too complex to handle it then such customers will always use the timeout policy via request options (where they may build and pass new policy instances based on dynamically changed settings). It is a workaround that may work for some use cases but no for others (customers that did not code for policy set in request options). It is always nice to not force customers to do rolling restarts to pick up a change.

Reply:
Discussed yesterday during our daily meeting - for now we keep it immutable - Bhaskar and I will look into options to make this mutable in a separate PR.